lib/assets/data_table/main.js

```diff
@@ -1,163 +1,310 @@
-                        <remix-icon :icon="data.content.order === 'asc' ? 'arrow-up-s-line' : 'arrow-down-s-line'"></remix-icon>
-                      </span>
-                    </div>
-                    <div v-if="column.type" class="table__cell-content-secondary">
-                      {{ column.type }}
-                    </div>
-                  </div>
-                </th>
-              </tr>
-            </thead>
-            <tbody class="table__body">
-              <tr v-for="row in data.content.rows" class="table__row table__row--hover table__row--no-trailing-border">
-                <td v-for="column in data.content.columns" :key="column.key" class="table__cell">
-                  {{ row.fields[column.key] }}
-                </td>
-              </tr>
-            </tbody>
-          </table>
-        </div>
+          <RiSearch2Line />
+        </button>
+      </span>
+    );
+  }
+
+  function Limit({ limit }) {
+    const handleChange = (event) => {
+      ctx.pushEvent("limit", { limit: parseInt(event.target.value) });
+    };
+
+    return (
+      <div>
+        <form>
+          <label className="input-label">Show</label>
+          <select className="input" value={limit} onChange={handleChange}>
+            <option value="10">10</option>
+            <option value="15">15</option>
+            <option value="20">20</option>
+            <option value={totalRows}>All</option>
+          </select>
+        </form>
       </div>
-      <div class="sr-only" role="region" id="table-info" aria-live="polite">
-        <span>{{ data.content.order_by ? orderDescription() : 'unordered'  }}</span>
-        <span>you are on table page {{ data.content.page }}</span>
+    );
+  }
+
+  function Pagination({ page, maxPage }) {
+    const prev = () => {
+      ctx.pushEvent("show_page", { page: page - 1 });
+    };
+
+    const next = () => {
+      ctx.pushEvent("show_page", { page: page + 1 });
+    };
+
+    return (
+      <div className="pagination">
+        <button
+          className="pagination__button"
+          onClick={prev}
+          disabled={page === 1}
+        >
+          <RiArrowLeftSLine />
+          <span>Prev</span>
+        </button>
+        <div className="pagination__info">
+          <span>
+            {page} of {maxPage || "?"}
+          </span>
+        </div>
+        <button
+          className="pagination__button"
+          onClick={next}
+          disabled={page === maxPage}
+        >
+          <span>Next</span>
+          <RiArrowRightSLine />
+        </button>
       </div>
-    `,
-
-    data() {
-      return {
-        data,
-        loading: false,
-      };
-    },
-
-    methods: {
-      prev() {
-        this.loading = true;
-        ctx.pushEvent("show_page", { page: this.data.content.page - 1 });
-      },
+    );
+  }
 
-      next() {
-        this.loading = true;
-        ctx.pushEvent("show_page", { page: this.data.content.page + 1 });
-      },
+  function App() {
```

Comment:
Maybe let's move all the components outside `init` and pass the necessary data/ctx as props to the `App` component? That should make things a bit more isolated and easier to analyse :)